src/IBM.WatsonDeveloperCloud.VisualRecognition.v3/VisualRecognitionService.cs

```diff
@@ -799,9 +799,6 @@ public SimilarImagesConfig FindSimilar(string collectionId, byte[] imageData, st
         #region Utils
         private string GetImageMimeTypeFromFilename(string filename)
         {
-            if (string.IsNullOrEmpty(filename))
-                throw new ArgumentNullException(nameof(filename));
-
             string imageMimeType = "";
             if (!string.IsNullOrEmpty(filename))

```

Comment:
I think that this "if" is unnecessary, since we have already tested the variable

Reply:
Yes, actually I think there are a lot of this type of check

```
if (somevar == null)
  throw new ArgumentNullException(nameof(somevar));
```

Should these be changed as well since if somevar is null we will get an exception trying to find the name?

Reply:
I understood, I had read wrong. It's okay.

Reply:
Ha, no I think you’re right! Maybe we should change the above code to use a string instead of nameof(somevar) since if somevar is null we won’t get the name

Reply:
I mean change the code in the generator so whenever we do a check for null variable, don’t try to get the name of the variable - just put a string. Something like “somevar is required for this method.”

Reply:
Ha, looking closer at this, this method is never used. But the point is good though! I regenerated to move from 

```
if (someVar == null)
  throw new ArgumentNullException(nameof(someVar));
```
to
```
if (someVar == null)
  throw new ArgumentNullException("'someVar' is required for 'someMethod()'");
```

Reply:
If “somevar” is null, the nameof will not throw an exception and using the nameof operator we can rename it

Reply:
Ah, good to know! It's not breaking so I will change in the generator next release.